From d375fbf2174016e37341bf59a6888870ea4cea14 Mon Sep 17 00:00:00 2001
From: Jakub Adam <jakub.adam@ktknet.cz>
Date: Sat, 1 Dec 2018 23:05:15 +0100
Subject: [PATCH 2/7] media: enable SIREN in SfB conferences

As reported by users and checked by myself, the issue with SIREN codec
in conferences has disappeared in Skype for Business and so SIPE can
start offering it during call negotiations again.

Thanks to Joakim Tjernlund for pointing this out.

Upstream-Status: Inappropriate [Not mine to merge]
Patch-URI: https://github.com/tieto/sipe/commit/d375fbf2174016e37341bf59a6888870ea4cea14.patch
---
 src/core/sipe-media.c | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/src/core/sipe-media.c b/src/core/sipe-media.c
index 7b70d86f..b9c7e3f4 100644
--- a/src/core/sipe-media.c
+++ b/src/core/sipe-media.c
@@ -430,6 +430,7 @@ static GList *
 get_local_codecs(struct sipe_media_call_private *call_private,
 		 struct sipe_media_stream_private *stream_private)
 {
+	struct sipe_core_private *sipe_private = call_private->sipe_private;
 	gboolean is_conference = g_strstr_len(SIPE_MEDIA_CALL->with,
 					      strlen(SIPE_MEDIA_CALL->with),
 					      "app:conf:audio-video:") != NULL;
@@ -445,12 +446,12 @@ get_local_codecs(struct sipe_media_call_private *call_private,
 		     * long, Communicator rejects such SDP message and does not
 		     * support the codec anyway. */
 		    sipe_strequal(name,"THEORA") ||
-		    /* For some yet unknown reason, A/V conferencing server
-		     * does not accept SIPE audio encoded with SIREN. We are
-		     * still able to decode incoming SIREN from server and with
-		     * MSOC client, bidirectional call using the codec works.
-		     * Until resolved, resort to PCMA or PCMU in conferences. */
-		    (is_conference && sipe_strequal(name,"SIREN"))) {
+		    /* For an unknown reason, MS Lync A/V conferencing server
+		     * does not accept SIPE audio encoded with SIREN. Disable
+		     * the codec when SIPE is connected to a server version
+		     * prior Skype for Business. */
+		    (!SIPE_CORE_PRIVATE_FLAG_IS(SFB) && is_conference &&
+		     sipe_strequal(name,"SIREN"))) {
 			GList *tmp;
 			sipe_backend_codec_free(codec);
 			tmp = i->next;
-- 
2.21.0

